src/sdl2/video.rs

```diff
@@ -885,38 +898,43 @@ impl WindowBuilder {
 
 impl Window {
     #[inline]
-    pub fn raw(&self) -> *mut ll::SDL_Window { self.raw }
+    pub fn raw(&self) -> *mut ll::SDL_Window { self.context.raw }
 
     #[inline]
     pub unsafe fn from_ll(subsystem: VideoSubsystem, raw: *mut ll::SDL_Window) -> Window {
-        Window {
-            subsystem: subsystem,
-            raw: raw
-        }
+        let context = WindowContext {
+            subsystem: subsystem.clone(),
+            raw: raw,
+        };
+        Window { context: Rc::new(context) }
     }
 
     #[inline]
-    pub fn subsystem(&self) -> &VideoSubsystem { &self.subsystem }
+    pub fn subsystem(&self) -> &VideoSubsystem { &self.context.subsystem }
 
     /// Initializes a new `RendererBuilder`; a convenience method that calls `RendererBuilder::new()`.
     pub fn renderer(self) -> RendererBuilder {
```

Comment:
Added `impl From<Window> for RendererBuilder`.
I still have the `into_canvas()` function because the compiler was not smart enough to infer the type on all the examples so `into_canvas` is still a useful method.